drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2021 ITE Corporation. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_uart
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/uart.h>
+#include <kernel.h>
+#include <pm/device.h>
+#include <soc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(uart_ite_it8xxx2, LOG_LEVEL_ERR);
+
+#define DRV_CONFIG(dev) \
+	((const struct uart_it8xxx2_config *)(dev)->config)
+
+struct uart_it8xxx2_config {
+	uint8_t port;
+	/* GPIO handle */
+	const struct device *gpio_dev;
+	/* UART handle */
+	const struct device *uart_dev;
+};
+
+enum uart_port_num {
+	UART1 = 1,
+	UART2,
+};
+
+#ifdef CONFIG_PM_DEVICE
```

Comment:
If you add `depends PM_DEVICE` to the serial/Kconfig.it8xxx2 file, then I think you can remove the #ifdefs in this file.

Reply:
If add` depends PM_DEVICE` and CONFIG_PM_DEVICE is not enabled, uart_ite_it8xxx2.c will not be built.
But serial driver of ITE is still necessary, including the initialization of uart(will be moved here from soc.c later) and alternate of pinmux control(currently with hard code in pinmux.c).